.github/workflows/unwanted-dependencies-check.yml

```diff
@@ -0,0 +1,42 @@
+name: unwanted-dependencies-check
+
```

Comment:
can this yaml run on prow job? as we have an issue which want to migrate some actions workflows to prow

Reply:
There's a requirement to make this optional, cmiiw, I don't think prow is optional

Reply:
There's a requirement to make the job optional, is it possible in prow to always run it but not cancelling the other pipeline?

Reply:
I'm not quite sure, but prow can run multiple jobs and choose to make some jobs optional. we might need to investigate prow( but prow is quite a huge project)

Reply:
I have confirmed with @natasha41575, it is decided that for this PR we'll still stick to GH action, and if your PR for prow is already merged we can migrate it to prow

Reply:
my PR for prow is still in pending review, we should merge this first